client/src/App.js

```diff
@@ -77,12 +89,32 @@ export default function App() {
     </AppGrid>
   )
 
-  function handleSignIn() {
-    push('/profile')
+  function handleSubmit(newProfile) {
+    postUser(newProfile)
+      .then(user => {
+        setProfile(user)
+        setUserId(user._id)
+        push('/profile')
+      })
+      .catch(error => console.error(error))
   }
 
-  function handleSubmit(newProfile) {
-    setProfile({ ...newProfile, ...profile })
+  function handleOnLogin(logProfile) {
+    getUser(logProfile)
+      .then(user => {
+        setProfile(...user)
+        const logUser = [...user]
+        setUserId(logUser[0]._id)
```

Comment:
Is `user` an object? If so, you could probably write it like this:
```suggestion
        setUserId(user._id)
```

Reply:
I know, but somehow when I update the state (ohne spread '...') 'profile' with the 'user' object that mogoDB returns, it doesn't work and the 'user._id is undefined, so I had to treat it as an array with only one object inside and store it in a consta